feature/gribi/ate_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go

```diff
@@ -0,0 +1,454 @@
+//
+// The testbed consists of ate:port1 -> dut:port1,
+// dut:port2 -> ate:port2.
+//
+//   * ate:port1 -> dut:port1 subnet 192.0.2.0/30
+//   * ate:port2 -> dut:port2 subnet 192.0.2.4/30
+//
+//   * Destination network for traffic test: 198.51.100.0/24
+
+const (
+	ateDstNetEntryNonDefault = "198.51.100.0/24" // IP Entry to be injected in non-default network instance
+	ateDstNetEntryDefault    = "203.0.113.0/24"  // IP Entry to be injected in default network instance
+	nonDefaultVRF            = "VRF-1"           // Name of non-default network instance
+	clientAOriginElectionID  = 10
+	clientBOriginElectionID  = 9
+	nhIndex                  = 1
+	nhgIndex                 = 42
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv4Len: 30,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv4Len: 30,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: 30,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv4Len: 30,
+	}
+)
+
+// TestRouteRemovalNonDefaultVRFFlush test flush with the following operations
+// 1. Flush request from clientA (the primary client) non default VRF should succeed.
+// 2. Flush request from clientB (not a primary client) non default VRF should fail.
+// 3. Failover the primary role from clientA to clientB.
+// 4. Flush from clientB non default VRF should succeed.
+// 5. Flush from default VRF should not delete all the gRIBI objects like NH and NHG.
+func TestRouteRemovalNonDefaultVRFFlush(t *testing.T) {
+	ctx := context.Background()
+
+	dut := ondatra.DUT(t, "dut")
+	configureDUT(t, dut)
+
+	ate := ondatra.ATE(t, "ate")
+	ateTop := configureATE(t, ate)
+
+	ateTop.Push(t).StartProtocols(t)
+	configureNetworkInstance(t, dut)
+	configureDUT(t, dut)
+
+	// Configure the gRIBI client clientA with election ID of 10.
+	clientA := &gribi.Client{
+		DUT:                  dut,
+		Persistence:          true,
+		InitialElectionIDLow: clientAOriginElectionID,
+	}
+
+	defer clientA.Close(t)
+
+	t.Log("Establish gRIBI clientA connection with PERSISTENCE set to TRUE")
+	if err := clientA.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection for clientA could not be established")
+	}
+
+	// Configure the gRIBI client clientB with election ID of 9.
+	clientB := &gribi.Client{
+		DUT:                  dut,
+		Persistence:          true,
+		InitialElectionIDLow: clientBOriginElectionID,
+	}
+
+	defer clientB.Close(t)
+
+	t.Log("Establish gRIBI clientB connection with PERSISTENCE set to TRUE")
+	if err := clientB.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection for clientB could not be established")
+	}
+
+	t.Log("Inject an IPv4Entry for 198.51.100.0/24 into VRF-1, with its referenced NHG and NH in the default routing-instance pointing to ATE port-2")
+	// clientA is primary client.
+	injectEntries(ctx, t, dut, clientA, nonDefaultVRF, ateDstNetEntryNonDefault)
```

Comment:
This is optional: I generally avoid passing `t` variables around.
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

An alternative style is to have the functional
```
func injectEntries(ctx, dut, ...) error { ... /* return some error whenever there's t.Fatal etc */ }
```
And in the TestXXX function call it like
```
if err := injectEntries(); err != nil { t.Error(err) }
```
The benefit of this style is to easily distinguish which step emits the error, if there're multiple usage of `injectEntries`